src/v/cluster/partition.cc

```diff
@@ -637,10 +642,8 @@ partition::local_timequery(storage::timequery_config cfg) {
               cfg.time,
               _raft->log()->start_timestamp(),
               cfg.time);
-            if (
-              _cloud_storage_partition
-              && _cloud_storage_partition->is_data_available()
-              && may_answer_from_cloud) {
+
+            if (may_answer_from_cloud) {
```

Comment:
@WillemKauf @andrwng

maybe it's too pedantic, but the original condition had `is_data_available` check and `may_answer_from_cloud` doesn't have that (at least not explicitly).

is that not important or its covered implicitly?

Reply:
`may_answer_from_cloud` is defined (this definition is unchanged in my PR) [here](https://github.com/WillemKauf/redpanda/blob/2823c0919ffcc13e0357632958c866bcb6fb42f3/src/v/cluster/partition.cc#L507) as:

```cpp
const bool may_answer_from_cloud = may_read_from_cloud() && _cloud_storage_partition->bounds_timestamp(cfg.time);
```

where `may_read_from_cloud()` is defined [here](https://github.com/WillemKauf/redpanda/blob/2823c0919ffcc13e0357632958c866bcb6fb42f3/src/v/cluster/partition.cc#L536) as 

```cpp
bool partition::may_read_from_cloud() const {
    return is_remote_fetch_enabled()
           && (_cloud_storage_partition && _cloud_storage_partition->is_data_available());
}
```

The condition you are highlighting here was previously written as 

```cpp
if(_cloud_storage_partition && _cloud_storage_partition->is_data_available() && may_answer_from_cloud)
```

or, expanded using the definition of `may_answer_from_cloud`,

```cpp
if(_cloud_storage_partition && _cloud_storage_partition->is_data_available() && _cloud_storage_partition->bounds_timestamp(cfg.time)) && may_read_from_cloud() 
```

and expanded using the definition of `may_read_from_cloud()`,

```cpp
if(_cloud_storage_partition && _cloud_storage_partition->is_data_available() && _cloud_storage_partition->bounds_timestamp(cfg.time)) && is_remote_fetch_enabled() && (_cloud_storage_partition && _cloud_storage_partition->is_data_available())
```

So, the condition as originally written had some superfluous checks. I should have called this out better in the PR or commit message, but this is a non-functional change.

Reply:
Thanks for the excellent derivation here. Even though the original may_answer_from_cloud 
 condition has superfluous checks at a functional level, does the new version of the condition preserve the necessary meanings transitively which will allow other code to evolve?

What I mean with an example is, are `may_read_from_cloud` and `_cloud_storage_partition->bounds_timestamp` defined well enough that they could be changed without knowing about may_answer_from_cloud while retaining the requirements of may_answer_from_cloud?

Reply:
As the non-original writer of that code, and re-reading it over, I would argue yes, `may_read_from_cloud()` is defined exactly as it should be defined, as is `bounds_timestamp()`. I don't think any future modifications to those functions should fundamentally change the questions they are trying to answer, those being

a) do we have the permission to read from cloud storage, and furthermore, do we have data available 
b) does our remote partition have data with the necessary timestamp(s) required to fulfill a timequery

respectively.

Maybe there could be another abstracted function to encapsulate the combination of those two such that `may_answer_from_cloud()` exists beyond a variable, but I would like to hear what you have to think about all this, since it seems largely philosophical/up for debate.

Reply:
thanks, this all makes sense. i don't really have any opinions. just noticed the change and thought it may have deserved a comment on the change, so just started the discussion. i think everything looks good. thanks for these detailed discussions!